src/analysis/processing/qgsoverlayutils.h

```diff
@@ -38,6 +41,11 @@ namespace QgsOverlayUtils
 
   void difference( const QgsFeatureSource &sourceA, const QgsFeatureSource &sourceB, QgsFeatureSink &sink, QgsProcessingContext &context, QgsProcessingFeedback *feedback, int &count, int totalCount, DifferenceOutput outputAttrs );
 
+  void intersection( const QgsFeatureSource &sourceA, const QgsFeatureSource &sourceB, QgsFeatureSink &sink, QgsProcessingContext &context, QgsProcessingFeedback *feedback, int &count, int totalCount, const QList<int> &fieldIndicesA, const QList<int> &fieldIndicesB );
+
+  QList<int> fieldNamesToIndices( const QStringList &fieldNames, const QgsFields &fields );
```

Comment:
Any change you can move this and indicesToFields to QgsProcessingUtils (and add tests)? There's other (non-overlay) algs which do the same things, and I'd like to reuse this code.

Reply:
Sure - moved in 091ed91 and tests added in 5e7719c